### PR TITLE
Finish up circuit drawing, expose to python

### PR DESCRIPTION
**Pull request by amccaskey**
Enable circuit drawing apart from library mode. Supersedes #1201 , picked up that work to help out @boschmitt on paternity leave.

### CODE REVIEW

File: python/runtime/cudaq/algorithms/py_draw.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#include <pybind11/complex.h>
+#include <pybind11/stl.h>
+
+#include "utils/OpaqueArguments.h"
+#include "cudaq/algorithms/draw.h"
```
Comment (schweitzpgi) [inline python/runtime/cudaq/algorithms/py_draw.cpp]: Whenever I see includes in this order, I immediately think it's a bug. Guess I've been doing LLVM for too long.

At any rate, the LLVM argument of always including local sources first is a good one as it prevents one from writing broken .h files that have dependences on other headers that the file doesn't include itself.

File: python/tests/unittests/test_kernel_builder.py [hunk 2]
```diff
@@ -978,6 +978,71 @@ def test_fermionic_swap_op():
...
+┤●      ├┤ ry(3.142) ├───────●──────┤ ry(-3.142) ├
+│  swap │├───────────┴╮╭───────────╮╰────────────╯
+┤●      ├┤ rz(-3.142) ├┤ rz(3.142) ├──────────────
+│       │╰────────────╯╰───────────╯              
+┤>      ├─────────────────────────────────────────
+╰───────╯                                         
+'''
+
+    # FIXME add this later
+    # assert circuit == expected_str
```
Comment (bettinaheim) [inline python/tests/unittests/test_kernel_builder.py:1045]: Looks like there is an assert missing.

**Review by bettinaheim — COMMENTED** [review]
In principle, lgtm, but there is a todo in the test to enable it.

File: python/runtime/cudaq/algorithms/py_draw.cpp (hunk 1 above)
Comment (amccaskey) [reply python/runtime/cudaq/algorithms/py_draw.cpp]: Will fix this

File: python/tests/unittests/test_kernel_builder.py (hunk 2 above)
Comment (amccaskey) [reply python/tests/unittests/test_kernel_builder.py:1045]: Turns out we had a bug in lowering S<adj> and T<adj> to QIR, and that's why the I couldn't get the strings to equal each other.

Comment (bettinaheim) [reply python/tests/unittests/test_kernel_builder.py]: Good find!

**Review by bettinaheim — APPROVED** [review]
Thanks!

**Pull request merged**